Import previous project assets

FerchusGames opened the pull request:
Used assets from my previous project "TowerFrog" to get this project running quickly.

FerchusGames committed:
Renamed "run" observables, objects and methods to "run"

The "run" action only implements the lateral movement of the player

FerchusGames committed:
Added properties to PlayerProperties for game feel

Though not yet implemented, these properties will make the character feel more responsive and fun to control

FerchusGames committed:
Added gravity properties to PlayerProperties

To control how gravity changes during different player actions

FerchusGames committed:
Added grid and tileset from TowerFrog into Main scene

FerchusGames committed:
Moved asset pack used files into _Main folder

FerchusGames committed:
Added reference script "PlayerMovement" from TowerFrog to implement movement.

FerchusGames committed:
Added Run acceleration

Horothenic commented on Towerfall/Assets/_Main/Scripts/Player/PlayerController.cs:
```diff
@@ -8,16 +8,23 @@ namespace Towerfall.Controllers
     {
+        void SetPlayerControllerData(IPlayerControllerData playerControllerData);
+
         IObservable<Unit> DashEnd { get; }
```
This should be in `IPlayerControllerInput`

FerchusGames replied:
c0da1b18286545cda7795908f1264d35c13d5a1f

Horothenic replied:
You moved the wrong one, I meant `DashEnd`

FerchusGames replied:
1637e433fda884bfa8b4a7739c561368b44cbd5a

FerchusGames committed:
Moved SetPlayerControllerData to IPlayerControllerInput

FerchusGames committed:
Refactored InstallBindings into multiple smaller functions

FerchusGames committed:
Refactored PlayerProperties into multiple interfaces and ScriptableObjects

FerchusGames committed:
Fixed errors in PlayerManager from mismatching properties names

FerchusGames committed:
Changed IPlayerProperties from having interfaces variables to having ScriptableObjects

FerchusGames committed:
Fixed PlayerProperties and modules' CreateAssetMenu

All modules had the same menuName and the PlayerProperties' one was inaccesible

FerchusGames committed:
Moved IPlayerProperties injection to base class

FerchusGames committed:
Created Properties assets

FerchusGames committed:
PlayerProperties now injects the interfaces instead of using the implementation classes

FerchusGames committed:
PlayerController and PlayerManager refactoring to make it more readable

FerchusGames commented:
I added a few commits to try to make the PlayerProperties more modular and to clean up the PlayerManager and PlayerController.

Horothenic commented on Towerfall/Assets/_Main/Scripts/Player/PlayerManager.cs:
```diff
@@ -64,70 +67,109 @@ public partial class PlayerManager : ITickable
             {
-                accelerationRate *= _playerProperties.JumpHangAccelerationMultiplier;
-                targetSpeed *= _playerProperties.JumpHangMaxSpeedMultiplier;
+                return _playerProperties.Movement.RunAccelerationRate 
+                       * _playerProperties.Movement.AirAccelerationMultiplier;
```
Why in 2 lines?

Horothenic commented on Towerfall/Assets/_Main/Scripts/Player/PlayerManager.cs:
```diff
@@ -64,70 +67,109 @@ public partial class PlayerManager : ITickable
+                       * _playerProperties.Movement.AirAccelerationMultiplier;
             }
+            
+            return _playerProperties.Movement.RunDecelerationRate 
+                   * _playerProperties.Movement.AirDecelerationMultiplier;
```
Same

FerchusGames committed:
Moved DashEnd to IPlayerControllerInput and SetPlayerControllerData back to IPlayerControllerEvent

FerchusGames committed:
Reformated lines that exceeded max character length
